src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,12 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
We should limit possibility to make Machine producer parametrized with Cake or Bird
```suggestion
public interface MachineProducer<T extends Machine> {
```